### ISSUE TITLE
Phase Transport Pipes

### ISSUE THREAD

**Issue by ReportBot9009** [tcooc/AdditionalPipesBC#119]
The phase Transport Pipes currently for BC7.0.6 and Compat7.0.5 with Forge 1407 are currently deleting items when going through them. Also, my old Logistics Pipes network I used to setup with Phase pipes is currently not working either. I could set down a phase pipe and a request pipe/table and request items wirelessly. The phase pipes don't seem to connect to the newest LPs. Currently on 0.9.1.11 for BC7.

I've tried all manner of Send/Receive functions and nothing seems to work. I sent a stack of dirt and only received 2. Items one at a time are deleted and lost in the pipe. They do not appear after breaking.

Thanks for keeping this mod alive though. I've loved and still love it so much! :)

**Comment by multiplemonomials** [comment]
Can you tell me about your pipe network?  How many pipes are there, what modes are they in, and are they connected to anything?

Also, the logistics pipes thing is a known issue.  See [here](https://github.com/RS485/LogisticsPipes/issues/348)

**Comment by ReportBot9009** [comment]
Yes. The Buildcraft to Phase pipe to chest is a simple setup testing the phase transport. Here is a picture: http://imgur.com/1lKou5o
Here you can see the redstone engine has had enough time to get to green and there is no cobblestone showing up in the other chest. I've set the the left phase to Send/Private/10 and the pipe on the right to Receive/Private/10. I've tried send/receive on both pipes as well as public and private. Nothing seems to work.

Here is my Logistics Pipe setup as I'm sure you're aware of this problem:
http://imgur.com/NOs8yq6
This did use to work flawlessly!

Hope this helps!!

**Comment by multiplemonomials** [comment]
To be honest, I have never used Logistics Pipes and have absolutely no idea how it works.  I'm interested to find out, though.

**Comment by ReportBot9009** [comment]
Oh? Well, it is a very fun system. It's more old school feel like Buildcraft. Think of it as AE only visual, in a way. And instead of storage through disks, you do it through normal chests and you link those pipes attached to the chest with inventory management type modules, then you can link it all up into an interface that shows you everything you have stored. Then you get to see all the items zip around. It's definitely a lot of fun and gives a more visual response to "things" going to and from places, unlike Applied Energistics. Although, the LP team and AE team have worked together to make sure the two systems worth flawlessly together. It's really neat! Hook up an AE system to an LP system and see items come out of your disks and zip through pipes to machines and whatnot instead of just little lightning flashes of the AE cables ;)

It's fun, hopefully you have a chance to check it out. Also for the sake of getting the phase pipe to play nicely with the LP network, heehee ;)... haha and now I'm rambling.

Thanks for the updates and keeping this mod alive, as always.

**Comment by multiplemonomials** [comment]
OK, I can reproduce your issue with that exact setup.  Oddly enough, I've never seen it before in testing the teleport pipes.

**Comment by ReportBot9009** [comment]
Phew, glad to know it wasn't just me and my modpack...

To be honest, I thought it may have been the gold pipes, but then I though, no, this would be the same across all pipes... I mean, I think it would be...

**Comment by multiplemonomials** [comment]
Ulp, I spoke too soon.  At first, I saw the items come out on the other end, travel a pipe or two, and then disappear.  When I used the debugger, though, it started working, and kept working.  I'm starting to think it was some sort of graphical glitch.

Did you see this too, or did the items never come out at all.

**Comment by ReportBot9009** [comment]
To be brutally honest, I let the system run down the entire 64 cobblestone. I averaged about 1 cobblestone in the chest for ~15. I then tried a creative engine that pulls a stack at a time (at least I think it does) and from a full stack pull two ended up in the chest.

**Comment by multiplemonomials** [comment]
But did you ever see items come out on the other end and disappear?

**Comment by ReportBot9009** [comment]
Yes. It would do 1 of 2 thing: Come out of the other end and disappear about 2 pipes later, or go through to the chest, which like I said, was very rare.

**Comment by ReportBot9009** [comment]
I forgot to say, most of the time, nothing would appear out of the other pipe.

**Comment by multiplemonomials** [comment]
I'll be honest, I'm kinda stumped here.  I haven't touched this code for a few versions, so it looks to me like something caused by a change in BuildCraft.  I'll keep looking.

**Comment by ReportBot9009** [comment]
I was in the Buildcraft IRC and asked if they knew about the problem. asie told me to update to the new BC 7.0.8 with Compat 7.0.7, but that didn't seem to fix the Item Phase pipes. The fluid and power Phase pipes work flawlessly, but the item pipes don't seem to be cooperating after the update...

Maybe you can head over there and ask for some help? I don't know how to code or else I would help out =*(...

**Comment by multiplemonomials** [comment]
I'll try, but I have final stuff this week, so probably nothing will happen until next weekend.

**Comment by ReportBot9009** [comment]
Hey, take your time. Priorities are priorities and your life is important. Minecraft isn't going away and neither is Buildcraft. Don't worry about it. 

Good luck in finals and hopefully we'll see some progress afterwards when you're ready!

Take care! :)

**Comment by ReportBot9009** [comment]
Hey, where have ya been? Hope finals went well. Are we going to see an update for this?

**Comment by multiplemonomials** [comment]
Sorry, I got kind of lazy and was working on other stuff.  As it happens, though, I *just* looked at this bug again like 2 hours ago, and finally figured out that it was caused by a bit of code that I commented out a long time ago (why? I have no idea.) that was keeping the server and client in sync.  I fixed it and commented it back in, and the bug is gone.  Sorry for the wait.  It should be fixed in the next rlease
